### PR TITLE
Add views for seeing outerbounds job compute costs

### PR DESCRIPTION
**Pull request by chelseatroy**
These are views for a dashboard that will allow decision-makers to track our compute costs on remotely run machine learning training jobs. The referenced tables have now been created in the bigquery console; after this merges, the next step is for me to expose these views for inclusion in Looker dashboards.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-4373)

### CODE REVIEW

**Review by BenWu — COMMENTED** [review]
I think these should go in the `monitoring` dataset instead of `monitoring_derived` since they're views.  They also wouldn't need the v1 suffix in that case, although you can use it if you want

File: sql/moz-fx-data-shared-prod/monitoring_derived/outerbounds_cost_per_flow_v1/view.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    2
+  ORDER BY
+    3 DESC
+)
+SELECT
+  flow_name,
+  SUM(cost_usd) AS total_cost_usd,
+  COUNT(flow_name) AS num_runs
+FROM
+  cost_by_run
```
[the remark this answers is not in the document]
Comment (chelseatroy) [reply sql/moz-fx-data-shared-prod/monitoring_derived/outerbounds_cost_per_flow_v1/view.sql:29]: You know what, you're right—originally I did think they'd diverge, but actually they shouldn't. I'll make this change and remove the CTE.

**Comment by chelseatroy** [comment]
@BenWu on `monitoring` vs `monitoring_derived`; happy to move it, want to first get a solid consensus on where this is supposed to go. @gkaberere thoughts?

**Comment by chelseatroy** [comment]
@BenWu for now I'm leaving it in `monitoring_derived`, but accepted your suggestions and also did some cleanup to remove the (now unused) CTE and group by column names!

**Comment by chelseatroy** [comment]
@edugfilho I think the failure here is that same `udf.sql::glam.histogram_cast_struct` test. I know you issued a fix for this; could you please let me know when it merges? I'll update once that's done 😺

**Review by BenWu — APPROVED** [review]
LGTM pending the test fixes from the other pr

**Comment by edugfilho** [comment]
sorry for this whole thing. @lelilia removed the failing test on https://github.com/mozilla/bigquery-etl/pull/5971 and I updated this branch, so it should pass now.

File: sql/moz-fx-data-shared-prod/monitoring_derived/outerbounds_cost_per_flow_run_v1/view.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
+CREATE OR REPLACE VIEW
+  `moz-fx-data-shared-prod.monitoring_derived.outerbounds_cost_per_flow_run_v1`
+AS
+SELECT
+  DATE_TRUNC(usage_start_time, DAY) AS invoice_day,
+  labels.value AS run_id,
+  IFNULL(SAFE.LEFT(labels.value, INSTR(labels.value, '-', -1) - 1), labels.value) AS flow_name,
+  SUM(cost) AS cost_usd
```
Comment (abhi-agg) [inline sql/moz-fx-data-shared-prod/monitoring_derived/outerbounds_cost_per_flow_run_v1/view.sql:8]: @chelseatroy We missed `credits.amount` in total cost (confirmed by @jasonthomas) like this:

```
SUM(cost) + SUM(IFNULL((SELECT SUM(c.amount)  FROM  UNNEST(credits) c), 0)) AS cost_usd
```
as per https://cloud.google.com/billing/docs/how-to/bq-examples#costs-details-per-invoice.

**Pull request merged**